src/dag_algo/mod.rs

```diff
@@ -637,3 +642,86 @@ pub fn collect_bicolor_runs(
 
     Ok(block_list)
 }
+
+/// Returns the transitive reduction of a directed acyclic graph
+///
+/// The transitive reduction of G = (V,E) is a graph G- = (V,E-)
+/// such that for all v,w in V there is an edge (v,w) in E- if and only if (v,w) is in E
+/// and there is no path from v to w in G with length greater than 1.
+///
+/// :param PyDiGraph graph: A directed acyclic graph
+///
+/// :returns: a directed acyclic graph representing the transitive reduction, and a map containing the index of a node in the original graph mapped to its equivalent in the resulting graph.
+/// :rtype: Tuple[:class:`~rustworkx.PyGraph`, dict]
```

Comment:
I think sphinx is smart enough to resolve this from `rtype`:

```suggestion
/// :rtype: Tuple[PyGraph, dict]
```